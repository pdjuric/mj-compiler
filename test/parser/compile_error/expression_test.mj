program function_call_test

class A { }

int var;
A a1, a2;

{
    void main() {

        //ugnezdeni pozivi

        if (5 < 'c') var++;
        if (5 == true) var++;
        if (a1 > a2 || a1 < a2) var++;
        if (true == true) var++;                // ok ?
        print(var);

        var = 'c' + 'c';
        var = true * false * true;
        var = a1 + a2;

   }
}