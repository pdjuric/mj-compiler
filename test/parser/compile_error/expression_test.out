ERROR 16:46:11,201 - line  13 - UncompatibleExprInRelop
INFO  16:46:11,201 - line  13 - Access to global variable var [Var var: int, 0, 0 ].
ERROR 16:46:11,202 - line  14 - UncompatibleExprInRelop
INFO  16:46:11,202 - line  14 - Access to global variable var [Var var: int, 0, 0 ].
INFO  16:46:11,279 - line  15 - Access to global variable a1 [Var a1: Class A, 1, 0 ].
INFO  16:46:11,279 - line  15 - Access to global variable a2 [Var a2: Class A, 2, 0 ].
ERROR 16:46:11,280 - line  15 - RefTypeRelopError
INFO  16:46:11,280 - line  15 - Access to global variable a1 [Var a1: Class A, 1, 0 ].
INFO  16:46:11,281 - line  15 - Access to global variable a2 [Var a2: Class A, 2, 0 ].
ERROR 16:46:11,281 - line  15 - RefTypeRelopError
INFO  16:46:11,281 - line  15 - Access to global variable var [Var var: int, 0, 0 ].
INFO  16:46:11,281 - line  16 - Access to global variable var [Var var: int, 0, 0 ].
INFO  16:46:11,281 - line  17 - Access to global variable var [Var var: int, 0, 0 ].
INFO  16:46:11,282 - line  17 - Call to global function print.
INFO  16:46:11,282 - line  19 - Access to global variable var [Var var: int, 0, 0 ].
ERROR 16:46:11,282 - line  19 - Expression term is not an int
ERROR 16:46:11,282 - line  19 - Expression term is not an int
INFO  16:46:11,282 - line  20 - Access to global variable var [Var var: int, 0, 0 ].
ERROR 16:46:11,282 - line  20 - Term factor is not an int
ERROR 16:46:11,283 - line  20 - Term factor is not an int
ERROR 16:46:11,283 - line  20 - Term factor is not an int
INFO  16:46:11,283 - line  21 - Access to global variable var [Var var: int, 0, 0 ].
INFO  16:46:11,283 - line  21 - Access to global variable a1 [Var a1: Class A, 1, 0 ].
INFO  16:46:11,283 - line  21 - Access to global variable a2 [Var a2: Class A, 2, 0 ].
ERROR 16:46:11,283 - line  21 - Expression term is not an int
ERROR 16:46:11,284 - line  21 - Expression term is not an int